### PR TITLE
Remove `robot_get_type` API

### PR DESCRIPTION
**Pull request by ad-daniel**
**Description**
Fixes https://github.com/cyberbotics/webots/issues/4121

As mentioned, this API no longer serves a tangible purpose and therefore can be removed

### CODE REVIEW

File: docs/reference/changelog-r2022.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
 ## Webots R2022a Revision 1
 Released on XX XX, 2022.
-
+  - Cleanup
+    - Removed API `wb_robot_get_type` as it no longer serves a purpose ([#4125](https://github.com/cyberbotics/webots/pull/4125)).
```
Comment (omichel) [inline docs/reference/changelog-r2022.md:7]: Not sure about the English here, what about:
```suggestion
    - Removed `wb_robot_get_type` API function as it no longer serves a purpose ([#4125](https://github.com/cyberbotics/webots/pull/4125)).
```

File: src/controller/matlab/mgenerate.py [hunk 2]
```diff
@@ -407,6 +404,8 @@ def main(args=None):
     gen(PROC, "wb_robot_set_data(data)", "robot")
     gen(FUNC, "wb_robot_wwi_receive_text()", "robot")
     gen(PROC, "wb_robot_wwi_send_text(text)", "robot")
+    gen(FUNC, "wb_robot_get_number_of_devices()", "robot")
+    gen(FUNC, "wb_robot_get_device_by_index(index)", "robot")
```
Comment (omichel) [inline src/controller/matlab/mgenerate.py:408]: Maybe one day, we can re-order this list in alphabetical order.

**Review by omichel — APPROVED** [review]
Thank you.
That looks good to me.

**Review by omichel — APPROVED** [review]
Thank you.

**Comment by ad-daniel** [comment]
Sorry, I wanted to double-check with fresh eyes. Should be good now.

**Review by omichel — APPROVED** [review]

**Pull request merged**